app/services/flags/create_flag.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class Flags::CreateFlag
+  include Service::Base
+
+  contract
+  policy :invalid_access
+  model :flag_instance, :instantiate_flag
+
+  transaction do
+    step :create
+    step :log
+  end
+
+  class Contract
+    attribute :name, :string
+    attribute :description, :string
```

Comment:
We should add a max length validation for these both here and in the model (and in the UpdateFlag service too) `length: { maximum: blah }`.  I think 200 characters for the name and 1000 characters for the description is reasonable to star.

Reply:
Also we should have validators to check for presence of both of these